### PR TITLE
Fe person uploader

### PR DESCRIPTION
**Pull request by salammohamadi**
### Working on the uploader, adding default file list for initial rendering adding missed uploadExtraInfo
******PLEASE DO NOT MERGE YET******

@da-saman

### CODE REVIEW

**Comment by da-saman** [comment]
temp WR do not show already added files

File: src/utils/schemas/personSchema.ts [hunk 1]
```diff
@@ -4,23 +4,22 @@ import * as yup from 'yup';
 export const personSchema = yup.object().shape({
   firstName: yup.string().required('نام الزامی می باشد'),
   lastName: yup.string().required('نام خانوادگی الزامی می باشد'),
-  nationalCode: yup.string().typeError(enterNumberMsg).required('کد ملی الزامی می باشد'),
-  phone: yup.number().typeError(enterNumberMsg).nullable().default(null),
-  email: yup.string().nullable().default(null),
-  fatherName: yup.string().nullable().default(null),
+  nationalCode: yup.string().required('کد ملی الزامی می باشد'),
```
Comment (da-saman) [inline src/utils/schemas/personSchema.ts:7]: why national code is string?

File: src/constants/wrConstants.ts [hunk 2]
```diff
@@ -144,3 +144,5 @@ export const approveData = (availableData:WRAvailable[], wr: CartableResponse):{
 
   return { disabled, massage };
 };
+
+export const cartableFilterFields:(keyof CartableResponse)[] = ['id', 'child', 'parentId', 'repairUnitId', 'requestImportanceId', 'workRequestAttachments'];
```
Comment (da-saman) [inline src/constants/wrConstants.ts:148]: cartableHiddenFields

File: src/constants/tables/workRequests/workRequestPMFormData.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ export const workRequestPMFormFields:FormFieldsDictionary<WorkRequestPMRequest>
     uploaderExtraInfo: {
       ...uploaderTypes[UPLOADER.SM_SINGLE_IMG_UPLOADER],
       uploadPath: 'WorkRequestPM',
+      multiple: true,
```
Comment (da-saman) [inline src/constants/tables/workRequests/workRequestPMFormData.ts:27]: SM_MULTI_IMG_UPLOADER

File: src/constants/tables/workRequests/workRequestCMFormData.ts [hunk 4]
```diff
@@ -26,6 +26,7 @@ export const workRequestCMFormFields:FormFieldsDictionary<WorkRequestCMRequest>
     uploaderExtraInfo: {
       ...uploaderTypes[UPLOADER.SM_SINGLE_IMG_UPLOADER],
       uploadPath: 'WorkRequestCM',
+      multiple: true,
```
Comment (da-saman) [inline src/constants/tables/workRequests/workRequestCMFormData.ts:29]: SM_MULTI_IMG_UPLOADER

File: src/components/shared/filePreview/FilePreview.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        ))}
+
+      </Image.PreviewGroup>
+    );
+  }
+
+  return (
+    <>
+      { initialFileList.map((file) => (
+        <EquipmentFilePreview key={file.uid} onClick={() => window.open(`${OpenAPI.BASE}/${file.url}`)}>
```
Comment (da-saman) [inline src/components/shared/filePreview/FilePreview.tsx:51]: EquipmentFilePreview in. shared folder?
in FilePreview?

File: src/utils/schemas/personSchema.ts (hunk 1 above)
Comment (salammohamadi) [reply src/utils/schemas/personSchema.ts:7]: add a test to test if it is a valid 10 digit number but still uses string value at the end

File: src/constants/wrConstants.ts (hunk 2 above)
Comment (salammohamadi) [reply src/constants/wrConstants.ts:148]: renamed

File: src/constants/tables/workRequests/workRequestPMFormData.ts (hunk 3 above)
Comment (salammohamadi) [reply src/constants/tables/workRequests/workRequestPMFormData.ts:27]: replaced with SM_MULTI_FILE_UPLOADER
multi file uploader

File: src/components/shared/filePreview/FilePreview.tsx (hunk 5 above)
Comment (salammohamadi) [reply src/components/shared/filePreview/FilePreview.tsx:51]: I haven't used this component yet. The plan is to use it inside DataTableColumn for table uplaod fields.
I was waiting for merging this branch and have enough data available in tables.

File: src/utils/schemas/personSchema.ts (hunk 1 above)
Comment (da-saman) [reply src/utils/schemas/personSchema.ts]: ok

File: src/components/shared/uploader/Uploader.tsx [hunk 6]
```diff
@@ -5,17 +5,20 @@ import {
...
+import { beforeUploadCheck } from '@constants/uploaders';
 import { UploaderContainerStyled as UploaderContainer } from './Uploader.styles';
 import { FileUploaderBig, ImageUploaderBig, UploadFile as UploadFileIcon } from '../../../assets/icons';
 
 const { Title } = Typography;
 
 interface IProps {
   title: string;
-  initialValue: string;
+  initialValue?: string | EquipmentFileUploadRes[]
```
Comment (da-saman) [inline src/components/shared/uploader/Uploader.tsx:21]: can we have a string[] instead of string | EquipmentFileUploadRes[]?

Comment (salammohamadi) [reply src/components/shared/uploader/Uploader.tsx]: backend sends a string for a single uploader but an array of { fileUrls, title, size } for multi ones
so filed.value (AKA initislValue) is string or mentioned array

File: src/constants/tables/workRequests/workRequestCMFormData.ts (hunk 4 above)
Comment (salammohamadi) [reply src/constants/tables/workRequests/workRequestCMFormData.ts]: thsnks

File: src/components/shared/uploader/Uploader.tsx [hunk 7]
```diff
@@ -25,98 +28,97 @@ export const Uploader = (props: React.PropsWithChildren<IProps>) => {
...
-            />
-          </Col>
-          )}
-        </Row>
+        {uploaderExtraInfo.smallInput ? (
+          <ButtonStyled radius='4px' className='upload-btn'>
+            انتخاب فایل
+          </ButtonStyled>
+        ) : (
+          <Row justify='space-around' align='middle'>
```
Comment (da-saman) [inline src/components/shared/uploader/Uploader.tsx:98]: move this to another component, in Uploader folder

**Pull request merged**